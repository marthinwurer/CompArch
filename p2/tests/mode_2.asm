;
; tests addressing mode 2
;
; uses instructions MOV, HALT, BPT
;
; uses other addressing modes 0, 6
;
	.globl	start
ptr:	.word	alpha
. = . + 50
alpha:	.word	1,2,4
;
. = . + 100
;
start:	mov	ptr,r2
	mov	#3,r1
	bpt
;
loop:	add	(r2)+,r3
	bpt
	sub	#1,r1
	bgt	loop
;
quit:	bpt
	halt
;
	.end	start
