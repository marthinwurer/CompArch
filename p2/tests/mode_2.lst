      1                                 ;
      2                                 ; tests addressing mode 2
      3                                 ;
      4                                 ; uses instructions MOV, HALT, BPT
      5                                 ;
      6                                 ; uses other addressing modes 0, 6
      7                                 ;
      8 000000                                   .globl  start
      9 000000  000052'                 ptr:     .word   alpha
     10 000052                          . = . + 50
     11 000052  000001  000002  000004  alpha:   .word   1,2,4
     12                                 ;
     13 000160                          . = . + 100
     14                                 ;
     15 000160  016702  177614          start:   mov     ptr,r2
     16 000164  012701  000003                   mov     #3,r1
     17 000170  000003                           bpt
     18                                 ;
     19 000172  062203                  loop:    add     (r2)+,r3
     20 000174  000003                           bpt
     21 000176  162701  000001                   sub     #1,r1
     22 000202  003373                           bgt     loop
     23                                 ;
     24 000204  000003                  quit:    bpt
     25 000206  000000                           halt
     26                                 ;
     27 000210  000160'                          .end    start


alpha    000052 R  
loop     000172 R  
pc     = 000007 %  
ptr      000000 R  
quit     000204 R  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000160 RG 
