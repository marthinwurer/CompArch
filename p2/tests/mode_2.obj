000 2 2a 00
02a 6 01 00 02 00 04 00
070 4 c2 1d 8c ff
074 4 c1 15 03 00
078 2 03 00
07a 2 83 64
07c 2 03 00
07e 4 c1 e5 01 00
082 2 fb 06
084 2 03 00
086 2 00 00
070
