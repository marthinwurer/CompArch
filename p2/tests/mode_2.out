CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 160
000160:  PS[00]  016702 177614  MOV 177614(R7),R2
000164:  PS[00]  012701  MOV (R7)+,R1
000170:  PS[00]  000003  BPT
  R0[000000]  R1[000003]  R2[000052]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000172]
000172:  PS[00]  062203  ADD (R2)+,R3
000174:  PS[00]  000003  BPT
  R0[000000]  R1[000003]  R2[000054]  R3[000001]
  R4[000000]  R5[000000]  R6[000000]  R7[000176]
000176:  PS[00]  162701  SUB (R7)+,R1
000202:  PS[00]  003373  BGT -12
000172:  PS[00]  062203  ADD (R2)+,R3
000174:  PS[00]  000003  BPT
  R0[000000]  R1[000002]  R2[000056]  R3[000003]
  R4[000000]  R5[000000]  R6[000000]  R7[000176]
000176:  PS[00]  162701  SUB (R7)+,R1
000202:  PS[00]  003373  BGT -12
000172:  PS[00]  062203  ADD (R2)+,R3
000174:  PS[00]  000003  BPT
  R0[000000]  R1[000001]  R2[000060]  R3[000007]
  R4[000000]  R5[000000]  R6[000000]  R7[000176]
000176:  PS[00]  162701  SUB (R7)+,R1
000202:  PS[04]  003373  BGT -12
000204:  PS[04]  000003  BPT
  R0[000000]  R1[000000]  R2[000060]  R3[000007]
  R4[000000]  R5[000000]  R6[000000]  R7[000206]
000206:  PS[04]  000000  HALT

Machine Halted - HALT instruction

Simulated time 118 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
